executor/insert_common.go

```diff
@@ -60,6 +60,15 @@ type InsertValues struct {
 	colDefaultVals  []defaultVal
 	evalBuffer      chunk.MutRow
 	evalBufferTypes []*types.FieldType
+
+	// cacheHasValue cache haveValue for consecutive autoid batch alloc.
+	cacheHasValue []bool
+	// Fill the autoID lazily to datum.
+	// This is used for being compatible with JDBC using getGeneratedKeys().
+	// By now in insert multiple values, TiDB can guarantee consecutive autoID in a batch.
+	lazyFillAutoID bool
```

Comment:
ditto.

Reply:
We need this to distinguish whether it is in the statement like `insert|replace values...`. Only this kind of statement needs to lazy allocate auto-id.